unstructured/ingest/connector/kafka.py

```diff
@@ -0,0 +1,315 @@
+                    # End of partition event
+                    logger.error(
+                        "%% %s [%d] reached end at offset %d\n"
+                        % (msg.topic(), msg.partition(), msg.offset())
+                    )
+            else:
+                collected.append(json.loads(msg.value().decode("utf8")))
+                if len(collected) >= num_messages_to_consume:
+                    logger.debug(f"Found {len(collected)} messages, stopping")
+                    running = False
+                    consumer.commit(asynchronous=False)
+                    break
+
+        return [
+            KafkaIngestDoc(
+                connector_config=self.connector_config,
+                processor_config=self.processor_config,
+                read_config=self.read_config,
+                raw_filename=msg["filename"],
+                raw_content=msg["content"],
+            )
+            for msg in collected
+        ]
+
+
+@dataclass
+class KafkaWriteConfig(WriteConfig):
+    batch_size: int = 50
+    num_processes: int = 1
+
+
+@dataclass
+class KafkaDestinationConnector(IngestDocSessionHandleMixin, BaseDestinationConnector):
+    """Connector to write BaseIngestDoc types to Kafka
+    Writes messages to Kafka in the format:
+    "type"<type>
+    "text":<the partitioned text>
+    "filename":<name of the upstream file>
+    """
+
+    write_config: KafkaWriteConfig
+    connector_config: SimpleKafkaConfig
+    _producer: t.Optional["Producer"] = None
+
+    @property
+    def kafka_producer(self):
+        if self._producer is None:
+            self._producer = self.create_producer()
+        return self._producer
+
+    def initialize(self):
+        pass
+
+    @requires_dependencies(["confluent_kafka"], extras="kafka")
+    def create_producer(self) -> "Producer":
+        from confluent_kafka import Producer
+
+        is_confluent = self.connector_config.confluent
+        bootstrap = self.connector_config.bootstrap_server
+        port = self.connector_config.port
+
+        conf = {
+            "bootstrap.servers": f"{bootstrap}:{port}",
+            "client.id": socket.gethostname(),
+        }
+
+        if is_confluent:
+            api_key = self.connector_config.access_config.kafka_api_key
+            secret = self.connector_config.access_config.secret
+            conf["sasl.mechanism"] = "PLAIN"
+            conf["security.protocol"] = "SASL_SSL"
+            conf["sasl.username"] = api_key
+            conf["sasl.password"] = secret
+
+        producer = Producer(conf)
+        logger.debug(f"Connected to bootstrap: {bootstrap}")
+        return producer
+
+    def check_connection(self):
+        try:
+            self.kafka_producer
+        except Exception as e:
+            logger.error(f"failed to validate connection: {e}", exc_info=True)
+            raise DestinationConnectionError(f"failed to validate connection: {e}")
+
+    @staticmethod
+    def chunks(iterable, batch_size=100):
```

Comment:
You actually already have this written as a generic util [here](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/ingest/utils/data_prep.py#L5)
```python
def chunk_generator(iterable, batch_size=100)
```

Reply:
Oh yeah. Now I remember that.